packages/deployer/subtasks/validate-namespaces.js

```diff
@@ -0,0 +1,49 @@
+const path = require('path');
+const rimraf = require('rimraf');
+const { subtask } = require('hardhat/config');
+const logger = require('@synthetixio/core-js/utils/logger');
+
+const {
+  getContractDataFromDebugFile,
+  getDebugDataFile,
+  getSlotAddress,
+  findDuplicateSlots,
+} = require('../internal/ast-helper');
+const { SUBTASK_VALIDATE_NAMESPACES } = require('../task-names');
+
+subtask(SUBTASK_VALIDATE_NAMESPACES).setAction(async (_, hre) => {
+  logger.subtitle('Validating namespaces');
+
+  const debugFiles = await _getDebugFiles(hre);
```

Comment:
Wouldn't this be better named as "namespaceFiles"?

Reply:
it's hardhat name for that file